neurodamus/utils/memory.py

```diff
@@ -179,11 +234,14 @@ def get_memory_usage(cls, count, synapse_type):
 
 class DryRunStats:
     _MEMORY_USAGE_FILENAME = "cell_memory_usage.json"
+    _ALLOCATION_FILENAME = "allocation.bin"
```

Comment:
```suggestion
    _ALLOCATION_FILENAME = "allocation.pkl"
```
At the very minimum. This seems to be a way more canonical ending. Also, can we shove that in a directory together with other simulation byproducts?